clang/include/clang-c/Dependencies.h

```diff
@@ -222,6 +222,17 @@ clang_experimental_DependencyScannerWorker_getFileDependencies_v1(
     CXDependencyScannerWorker Worker, int argc, const char *const *argv,
     const char *WorkingDirectory, CXModuleDiscoveredCallback *MDC,
     void *Context, CXString *error);
+
+/**
+ * Same as \c clang_experimental_DependencyScannerWorker_getFileDependencies_v1,
+ * but get the dependencies by module name alone.
+ */
+CINDEX_LINKAGE CXFileDependencies *
+clang_experimental_DependencyScannerWorkerByModName_getFileDependencies_v0(
```

Comment:
Continuing our discussion on naming from the original PR.

Yes, the `CXFileDependencies` type we're returning here does contain both `FileDeps` and `ModuleDeps`. However, I think its name and the name of the existing functions (`getFileDependencies`) don't refer to the contents of the result, but rather to the fact that they will contain/compute dependencies of a C/ObjC **file**.

So if we're scanning dependencies of a named module, I think it's reasonable to call the new function `getModuleDependencies` and consider renaming `CXFileDependencies` to simply `CXDependencies`.

@Bigcheese do you have any thoughts on this?

Reply:
I think the current naming we have is a bit confusing now. I think it would be better to have the following:

`CXDependencies`
`clang_experimental_DependencyScannerWorker_getDependenciesByModName_v0`
`clang_experimental_DependencyScannerWorker_getDependenciesByFile_v0`

Not sure if we should do this now as we'll need to update users, but it seems to make more sense.